src/datatypes/src/vectors/constant.rs

```diff
@@ -83,6 +83,37 @@ impl ConstantVector {
             self.length,
         )))
     }
+
+    pub(crate) fn take_vector(&self, indices: &UInt32Vector) -> Result<VectorRef> {
+        if indices.is_empty() {
+            return Ok(self.slice(0, 0));
+        }
+        ensure!(
+            indices.null_count() == 0,
+            error::UnsupportedOperationSnafu {
+                op: "taking a null index",
+                vector_type: self.vector_type_name(),
+            }
+        );
+
+        let len = self.len();
+        let arr = indices.to_arrow_array();
+        let indices_arr = arr.as_any().downcast_ref::<UInt32Array>().unwrap();
+        if !arrow::compute::min_boolean(
+            &arrow::compute::lt_scalar(indices_arr, len as u32).unwrap(),
+        )
+        .unwrap()
+        {
+            return Err(error::ArrowComputeSnafu.into_error(
+                arrow::error::ArrowError::ComputeError("Array index out of bounds".to_string()),
+            ));
+        }
```

Comment:
Looks like this require allocating an additional boolean array and makes getting index out-of-bounded inconvenient.  I'd prefer to use the original version that using a loop. @v0y4g3r

Reply:
> Looks like this require allocating an additional boolean array and makes getting index out-of-bounded inconvenient. I'd prefer to use the original version that using a loop. @v0y4g3r

I think either approach is ok, arrow compute may have better performance with simd enabled but brings more memory footprint. 

BTW I suggest just panic on out-of-bound access instead return an error, like `take_indices` of `PrimitiveVector`s.